braidpool_spec.md

```diff
@@ -460,33 +675,73 @@ formed, fully signed and valid bitcoin transactions that can be broadcast. See
 [Payout Authorization](#payout-authorization) for considerations on how to
 sign/authorize this UHPO transaction.
 
+[kp]
+
+The UHPO transactions are signed and broadcast using the braidpool P2P
+network. In case of the pool shutting down, the miners will have to figure out a
+communication channel to broadcast their signature shares and reach agreement on
+spending the UHPO.
+
+This sounds like a problem.
+
+[kp]
+
 We don't ever want to actually have to broadcast this UHPO set transaction
 except in the case of pool failure. Similar to other optimistic protocols like
 Lightning, we will withhold this transaction from bitcoin and update it
-out-of-band with respect to bitcoin. With each new block we will update the UHPO
-set transaction to account for any new shares since the last block mined by the
-pool.
+out-of-band with respect to bitcoin. With each new block found by braidpool, we
+will update the UHPO set transaction to account for any new shares since the
+last block mined by the pool.
 
 Furthermore a decentralized mining pool should support "withdrawal" by hashers.
-This would take the form of a special message or transaction sent to the pool
+This will take the form of a special message or transaction sent to the pool
 (and agreed by consensus within the pool) to *remove* a hasher's output from the
 UHPO set transaction, and create a new separate transaction which pays that
 hasher, [authorizes](#payout-authorization) it, and broadcasts it to bitcoin.
 Miners may not cash out for the *current* difficulty adjustment window (because
 the share/BTC price is not yet decided), but may only cash out for the *last*
 (and older) difficulty adjustment window(s).
 
+[kp]
+
+We need to address potential race condition and instability periods when miners
+are cashing out and new bitcoin blocks are being found. If there is no such
+instability, we should explain why. I expect this question will be raised during
+review.
+
+
+[/kp]
+
+
```

Comment:
A cash-out storm is expected at each DA epoch. This is partially mitigated by having 2 weeks in which market makers and miners can swap and aggregate their shares.

I'm not sure what else we can do about this. We could rate-limit withdrawal transactions, or just let the fee market take care of it.